packages/anoncreds/src/formats/LegacyIndyCredentialFormatService.ts

```diff
@@ -0,0 +1,603 @@
+import type { LegacyIndyCredentialFormat } from './LegacyIndyCredentialFormat'
+import type {
+  AnonCredsCredential,
+  AnonCredsCredentialOffer,
+  AnonCredsCredentialRequest,
+  AnonCredsCredentialRequestMetadata,
+} from '../models'
+import type { AnonCredsIssuerService, AnonCredsHolderService } from '../services'
+import type { AnonCredsCredentialMetadata } from '../utils/metadata'
+import type {
+  CredentialFormatService,
+  AgentContext,
+  FormatCreateProposalOptions,
+  FormatCreateProposalReturn,
+  FormatProcessOptions,
+  FormatAcceptProposalOptions,
+  FormatCreateOfferReturn,
+  FormatCreateOfferOptions,
+  FormatAcceptOfferOptions,
+  CredentialFormatCreateReturn,
+  FormatAcceptRequestOptions,
+  FormatProcessCredentialOptions,
+  FormatAutoRespondProposalOptions,
+  FormatAutoRespondOfferOptions,
+  FormatAutoRespondRequestOptions,
+  FormatAutoRespondCredentialOptions,
+  CredentialExchangeRecord,
+  CredentialPreviewAttributeOptions,
+  LinkedAttachment,
+} from '@aries-framework/core'
+
+import {
+  CredentialFormatSpec,
+  AriesFrameworkError,
+  IndyCredPropose,
+  JsonTransformer,
+  Attachment,
+  CredentialPreviewAttribute,
+  AttachmentData,
+  JsonEncoder,
+  utils,
+  MessageValidator,
+  CredentialProblemReportError,
+  CredentialProblemReportReason,
+} from '@aries-framework/core'
+
+import { AnonCredsError } from '../error'
+import { AnonCredsIssuerServiceSymbol, AnonCredsHolderServiceSymbol } from '../services'
+import { AnonCredsRegistryService } from '../services/registry/AnonCredsRegistryService'
+import {
+  convertAttributesToCredentialValues,
+  assertCredentialValuesMatch,
+  checkCredentialValuesMatch,
+  assertAttributesMatch,
+  createAndLinkAttachmentsToPreview,
+} from '../utils/credential'
+import { AnonCredsCredentialMetadataKey, AnonCredsCredentialRequestMetadataKey } from '../utils/metadata'
+
+const INDY_CRED_ABSTRACT = 'hlindy/cred-abstract@v2.0'
+const INDY_CRED_REQUEST = 'hlindy/cred-req@v2.0'
+const INDY_CRED_FILTER = 'hlindy/cred-filter@v2.0'
+const INDY_CRED = 'hlindy/cred@v2.0'
+
+export class LegacyIndyCredentialFormatService implements CredentialFormatService<LegacyIndyCredentialFormat> {
+  public readonly formatKey = 'indy' as const
+  public readonly credentialRecordType = 'anoncreds' as const
```

Comment:
For indy-sdk we can't access the record itself, we can just retrieve it with the indy-sdk. It just refers to that we have an anoncreds credential with the specific id (in the credential record binding). If you use the `indy-sdk` it will be stored in a different place, so in this case it's just the return value from AnonCredsHolderService.getCredential

Reply:
Hopes this makes sense

Reply:
Yeah thanks! Maybe a small comment might help :).